moecache.py

```diff
@@ -422,25 +478,39 @@ def set(self, key, val, exptime=0):
 
         if isinstance(val, str):
             flag = ' 274 '  # 18 | 0x100
-            sent = val
+            if Client.is_py3:
+                sent = bytearray(val, 'utf-8')
+            else:
+                sent = val
         else:
             flag = ' 256 '  # 0 | 0x100
             sent = pickle_dumps(val)
 
         buf = OStringStream()
-        buf.write('set ')
-        buf.write(key)
-        buf.write(flag)
-        buf.write(str(exptime))
-        buf.write(' ')
-        buf.write(str(len(sent)))
-        buf.write('\r\n')
-        buf.write(sent)
-        buf.write('\r\n')
+        if Client.is_py3:
+            buf.write(bytearray('set ', 'utf-8'))
+            buf.write(key_bytes)
+            buf.write(bytearray(flag, 'utf-8'))
+            buf.write(bytearray(str(exptime), 'utf-8'))
+            buf.write(bytearray(' ', 'utf-8'))
+            buf.write(bytearray(str(len(sent)), 'utf-8'))
+            buf.write(bytearray('\r\n', 'utf-8'))
+            buf.write(sent)
+            buf.write(bytearray('\r\n', 'utf-8'))
+        else:
+            buf.write('set ')
+            buf.write(key)
+            buf.write(flag)
+            buf.write(str(exptime))
```

Comment:
The whole switch is not needed.  `b` prefix works in python2.  For `str(exptime)`, `str(exptime).encode()` works in both versions.